pennylane/tape/tape.py

```diff
@@ -158,9 +158,21 @@ def stop_at(obj):  # pylint: disable=unused-argument
                     tape._obs_sharing_wires
                 )
             except (TypeError, ValueError) as e:
+                if any(
+                    m.return_type in (Probability, Sample, Counts, AllCounts)
+                    for m in tape.measurements
+                ):
+                    raise qml.QuantumFunctionError(
+                        "Only observables that are qubit-wise commuting "
+                        "Pauli words can be returned on the same wire.\n"
+                        "Try removing all probability, sample and counts measurements "
```

Comment:
Worth maybe to mention splitting execution, it is not really clear why it would work by removing some operations. Because it is possible to execute non commuting obs if we don't have specific measurements.

Reply:
Good idea 👍🏼